tests/query/test_query_ast.py

```diff
@@ -119,3 +121,32 @@ def replace(exp: Expression) -> Expression:
     assert list(query.get_all_expressions()) == list(
         expected_query.get_all_expressions()
     )
+
+
+def test_get_all_columns() -> None:
+    query_body = {
+        "selected_columns": [
+            ["f1", ["column1", "column2"], "f1_alias"],
+            ["f2", [], "f2_alias"],
+        ],
+        "aggregations": [
+            ["count", "platform", "platforms"],
+            ["uniq", "platform", "uniq_platforms"],
+            ["testF", ["platform", "field2"], "top_platforms"],
```

Comment:
I guess not all of arguments passed to the aggregation function will necessarily be columns, though we seem make the assumption that they are. Should we attempt to figure this out or would that be too difficult to determine?

Reply:
Added a function as well, seems it still works.

Reply:
Function works well because we have a distinct syntax for expressing functions in the Snuba query language. The ones I am concerned about is the function arguments that are neither functions nor columns. For example something like `["formatDateTime", ["timestamp", "%Y-%m-%d"], "formatted_time"]` I'm guessing would assume the 2nd arg is also a column since it's a string. And if it wasn't a string but something else like an int, I'm not sure what we should expect to happen. This might be tricky to do correctly, but I wonder if we should try since assuming they're all columns and they are not might lead to some weird behaviour.

Reply:
Actually we are fine with this, since literal strings are not confused with column names since they have to be quoted.

```
["formatDateTime", ["timestamp", "%Y-%m-%d"], "formatted_time"]
```

Is not a valid function call for snuba already today:

```
"message": "[47] DB::Exception: Missing columns: '%Y-%m-%d' while processing query: 'SELECT formatDateTime(timestamp, `%Y-%m-%d`) AS formatted_time FROM
```

To make it work we need that to be:
```
    "selected_columns": [["formatDateTime", ["timestamp", "'%Y-%m-%d'"], "formatted_time"]],
```

which the parser would turn into a Literal, not a column